crates/core/action-context/src/lib.rs

```diff
@@ -26,13 +26,59 @@ impl TargetState {
     }
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct TaskNamedMutexes {
+    mutexes: Arc<std::sync::Mutex<FxHashMap<String, Arc<tokio::sync::Mutex<()>>>>>,
```

Comment:
Mutexes within a mutex may be much. Maybe dashmap will work better here? https://crates.io/crates/dashmap

Reply:
Dashmap sounds pretty good, I admit my implementation is pretty naive. I'll try it to see if it behaves well 😃 .

Reply:
Done :) .